node/src/types/block/past_finality_signatures.rs

```diff
@@ -0,0 +1,221 @@
+use std::collections::BTreeSet;
+
+use casper_types::{
+    bytesrepr::{self, Bytes, FromBytes, ToBytes},
+    PublicKey,
+};
+
+use datasize::DataSize;
+use itertools::Itertools;
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+/// List of identifiers for finality signatures for a particular past block.
+#[derive(
+    Clone,
+    DataSize,
+    PartialEq,
+    Eq,
+    PartialOrd,
+    Ord,
+    Hash,
+    Serialize,
+    Deserialize,
+    JsonSchema,
+    Debug,
+    Default,
+)]
+pub struct PastFinalitySignatures(Vec<u8>);
+
+impl PastFinalitySignatures {
+    #[allow(dead_code)] //TODO remove it when the code is used in a next ticket
+    pub fn from_validator_set<'a>(
+        public_keys: &BTreeSet<PublicKey>,
+        all_validators: impl IntoIterator<Item = &'a PublicKey>,
+    ) -> Self {
+        let bits = all_validators
+            .into_iter()
+            .map(|key| public_keys.contains(key))
+            .chunks(8)
+            .into_iter()
+            .map(|bits| {
+                bits.enumerate()
+                    .map(|(i, bit)| u8::from(bit) << (7 - i))
+                    .fold(0, |acc, u| acc | u)
```

Comment:
This could be
```suggestion
                    .map(|(i, bit)| u8::from(bit) << (7 - i))
                    .sum()
```
or even
```suggestion
                    .fold(0, |accumulated, (index, is_present)| {
                        accumulated | (u8::from(is_present) << (7 - index))
                    })
```

Reply:
Done